docs/next-docs/code-conventions.md

```diff
@@ -23,3 +23,48 @@ TypeScript types should be defined inside `src/lib/types.ts` file. Note that som
 ### Interfaces
 
 TypeScript types should be defined inside `src/lib/interfaces.ts` file. Note that some pre-existent interfaces could be defined in other files and will be temporarily kept there during the migration, to facilitate synchronization. These interfaces should be moved to `src/lib/interfaces.ts` later.
+
+### Component Props
+
+For all components, use the following pattern:
+
+```tsx
+type ComponentProps = {
+  // Prop types
+}
+
+// Destructure props at the declaration and use the name `props` when using the spread operator for the remaining props not specifically needed
+const Component = ({ title, label, ...props }: ComponentProps) => {
+  // Component code
+}
+
+/**
+ * Components using `forwardRef` from the Chakra UI package
+ *
+ * The first argument of the generic types is the props type signature.
+ *
+ * For the second argument of the generic types, you are declaring the primary element type that the component will render.
+ * This could be a `div`, `span`, `button`, etc. or a custom component (typeof Button) if said component is being used in the return.
+ */
+const Component = forwardRef<ComponentProps, "div">(
+  ({ title, label, ...props }, ref) => {
+    // Component code
+  }
+)
+```
+
+#### Prop Type Naming Convention
+
+For the props type signature use the naming convention `<ComponentName>Props` to provide an explicit name for the type. This is helpful for importing the signature to other files like page component and not force the user to alias the import every time, while providing readability.
```

Comment:
Do we want to always export the Props type?

Reply:
Or are we cool doing this on an individual basis if it is needed elsewhere?

Reply:
I do not have a preference to argue for here.

Reply:
> Or are we cool doing this on an individual basis if it is needed elsewhere?

I think we were already doing that in the main repo, we were not enforcing the export.